### PR TITLE
Feature/books and labels.

### PR DESCRIPTION
**Pull request by ABDULALI3468**
### Team member #1 @ABDULALI3468 
- Create a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Label class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
        - should override the method from the parent class
        - should return true if the parent's method returns true OR if cover_state equals "bad"
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
    - List all books
    - List all labels (e.g. 'Gift', 'New')
    - Add a book
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - books table (add all properties and associations from the parent Item class as table columns)
    - labels table

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative 'item'
+
+class Book < Item
+  attr_accessor :publish, :cover_state
+
+  def initialize(archieved, publish, cover_state, publish_date)
```
Comment (teckim) [inline book.rb]: Kindly, check the typo: archieved, and it seems like the requirements said 'publisher' and not 'publish'.

File: cli.rb [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
+      Cover State: #{book['cover_state']} ,
+      Published Date: #{book['publish_date']}")
+    end
+  end
+
+  def list_labels
+    @labels.each_with_index do |label, index|
+      puts("(#{index})- Title: #{label['title']} , Color: #{label['color']}")
+    end
+  end
```
Comment (teckim) [inline cli.rb:50]: Do you think we can move this logic to the `main.rb` file, and use the `read_input` method to print a message and wait for user input.

File: item.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 class Item
-  attr_reader :id, :genre, :author, :source, :label, :publish_date, :archieved
+  attr_accessor :genre, :author, :source, :label, :publish_date
+  attr_reader :id, :archieved
 
-  def initialize(id, publish_date, archieved)
-    @id = id
+  def initialize(publish_date, archieved)
```
Comment (teckim) [inline item.rb]: It seems like the archieved typo appeared here also.

Comment (ABDULALI3468) [reply item.rb]: @teckim **I had resolved the Typos as requested by you.**

File: schema.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  COLOR VARCHAR
+);
+
+CREATE TABLE BOOKS(
+  ID INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  PUBLISHER VARCHAR,
+  COVER_STATE VARCHAR,
+  PUBLISH_DATE DATE,
+  ARCHIVED BOOLEAN,
+);
```
Comment (teckim) [inline schema.sql]: Good work on creating the `books` table, but the requirements mentioned: 
> books table (add all properties and associations from the parent Item class as table columns)

so I think we have to add all the properties of the `item` class

Comment (ABDULALI3468) [reply schema.sql]: @teckim No they mean the Properties of book class and ALL other properties you are inheriting as super or calling it in INITIALIZATION.

**Review by teckim — CHANGES_REQUESTED** [review]
Great work @ABDULALI3468, there are some changes that I think are worth checking. Kindly check the comments and feel free to ask if anything.

**Review by teckim — APPROVED** [review]

**Pull request merged**